### DISCUSSION TITLE
[EDIT] Account_id Param Required when submitting an order as a Guest

### DISCUSSION THREAD

**Discussion by m4tt133** [swellstores/community#93] [Help] [answered]
Hi!

I have been struggling with a strange error on my development storefront.

**The Problem:**
 It seems that I cannot place an order using the Frontend API when the user is not authenticated.

**Description:**
Although the default checkout hosted on Swell seems to be working, my application receives an error:

![Capture-order-console-error](https://user-images.githubusercontent.com/92744914/213515763-fe00d5bb-583a-4a0a-a2fc-de1cf77b9697.JPG)
![capture-error](https://user-images.githubusercontent.com/92744914/213515862-548aa59e-f8de-42c6-94b4-cb1c765ff03f.JPG)

**Checking if the swell session is present:**

![API Call - Checkout Last Step](https://user-images.githubusercontent.com/92744914/213514027-7053cced-f48c-46ee-8cac-49f1ba6fa35d.JPG)
![API Call 2](https://user-images.githubusercontent.com/92744914/213514019-d86174b0-12ca-47b8-a0cc-726c95ad6815.JPG)
![Capture-cookies](https://user-images.githubusercontent.com/92744914/213514764-cf1f8918-0a5c-4127-8065-c55b8d749e3b.JPG)

The API call = `const user = await swell.account.get();` resolves in null and the cookies seem to be cleared when I log out on my front end.

**Desired Outcome:**
Unauthenticated users can place an order.

Can you help me provide any clue in figuring this out? 

**EDIT:**

I think that I've found a cause for this problem. 

It seems that according to your Frontend documentation, the cart object for the unauthenticated user should have a property account_logged_in=true and guest=true, however, my cart object when the user is unauthenticated seems to have account_logged_in=null and guest=true. Is there a way that I can set this cart property to false, with the code? 

I think that that is why your API responds with a missing account_id error :)

Take a look at the screenshots below.

**Logged OUT:**
![loggedOut](https://user-images.githubusercontent.com/92744914/213915422-ca38257f-57d1-4a6d-917a-a0bdc8194b0b.JPG)

**Logged IN:**
![loggedIn](https://user-images.githubusercontent.com/92744914/213915423-471e94cb-b4b2-425c-9f3a-c0095b218a8f.JPG)

It has been a frustrating error that has been happening for about a month. (I have not been touching any code responsible for order placing since then.)

Cheers,

**Comment by logeshswell** [comment, marked as answer]
@m4tt133 What you see is expected behavior and the order can't be submitted without `account.email` (or account_id).
Please look into the below section to update email and password (if applicable) to the cart before submitting the order.
https://developers.swell.is/frontend-api/carts#update-cart-account-info

If you want your users to submit the order without authentication, then pass only email_id as follows:
```js
await swell.cart.update({
  account: {
    email: 'test@example.com'
  }
})
await swell.cart.submitOrder()
```
